src/pages/index.js

```diff
@@ -54,43 +58,246 @@ const addFormValidator = new FormValidator(
   addCardFormElement
 );
 
-section.renderItems();
-profileEditPopup.setEventListeners();
-addCardPopup.setEventListeners();
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
-imagePopup.setEventListeners();
+function handleDeleteClick(cardId) {
+  if (!cardId || typeof cardId !== "string") {
+    console.error("Invalid cardId for deletion:", cardId);
+    return;
+  }
+  deleteCardPopup.open({ cardId: cardId });
+}
 
 function handleImageClick(data) {
   imagePopup.open(data);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const sanitizedLink =
+    cardData.link.replace(/["'\s]+$/, "").trim() ||
+    "https://placehold.co/150x150";
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: sanitizedLink,
+      _id: cardData._id || null,
+    },
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+
   const element = card.getView();
+  if (!cardData._id) {
+    console.warn("Card created with no _id, deletion might fail:", cardData);
+  } else {
+    const initialIsLiked =
+      cardData.isLiked !== undefined ? cardData.isLiked : false;
+    likeStatuses.set(cardData._id, initialIsLiked);
+    const likeButton = element.querySelector(".card__like-button");
+    if (likeButton) {
+      likeButton.classList.toggle("card__like-button_active", initialIsLiked);
```

Comment:
Now only the `Card` class knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle